packages/axios-extension/src/auth/connection.ts

```diff
@@ -22,7 +22,7 @@ export class Cookies {
      * @returns cookies object
      */
     public setCookies(response: AxiosResponse): Cookies {
-        if (response.headers?.['set-cookie']) {
+        if (response?.headers?.['set-cookie']) {
```

Comment:
`response` shouldn't be optional here. The function must be called with a `response` object. If not, then there is a bug in the caller code, so I would prefer to remove the `?` here and if there are cases with `response` not being defined, then lets fix them on caller side.

Reply:
Reverted back in https://github.com/SAP/open-ux-tools/pull/929/files